lib/imagec/download.go

```diff
@@ -154,15 +154,16 @@ func (ldm *LayerDownloader) DownloadLayers(ctx context.Context, ic *ImageC) erro
 				defer topDownload.Transfer.Release(watcher)
 
 				ldm.registerDownload(topDownload)
-				cache.LayerCache().AddNew(id)
+				layer.Downloading = true
+				LayerCache().Add(layer)
```

Comment:
Done, also cleaned up some logging consistency issues and ordered the goroutines from layers -> image -> repo -> container, for aesthetic purposes.